tests/foreman/ui/test_host.py

```diff
@@ -129,13 +130,18 @@ def setUpClass(cls):
         cls.proxy.organization = [cls.org_]
         cls.proxy = cls.proxy.update(['organization'])
 
-        # Search for domain and associate org, location
+        # Search for existing domain or create new otherwise. Associate org,
+        # location and dns to it
         _, _, domain = settings.server.hostname.partition('.')
         cls.domain = entities.Domain().search(
             query={
                 u'search': u'name="{0}"'.format(domain)
             }
-        )[0]
+        )
+        if len(cls.domain) == 1:
+            cls.domain = cls.domain[0]
+        else:
+            cls.domain = entities.Domain().create()
         cls.domain_name = cls.domain.name
         cls.domain.location = [cls.loc]
```

Comment:
Could you also please use append for loc/org list instead of overwriting entire list? As we previously discussed, overwriting is not the greatest idea from test isolation perspective.